index.ts

```diff
@@ -0,0 +1,39 @@
+function fizzbuzz() {
+    for (let i = 1; i < 256; i++) {
+
+        let output: string[] = []
+        let rules = {3: "Fizz", 5: "Buzz", 7: "Bang", 11: "Bong", 13: "Fezz"}
+
+        const checkDivisor = (i: number, divisor: number): boolean => {
+            return i % divisor === 0
+        }
+
+        const applyRule = (i: number, divisor: number) => {
+            if (checkDivisor(i, divisor)) {
+                output.push(rules[divisor])
+            }
+        }
+
+        if (checkDivisor(i, 11)) {
+            applyRule(i, 13)
+            applyRule(i, 11)
+        }
+    
+        if (output.length === 0) {
+            applyRule(i, 3)
+            applyRule(i, 13)
+            applyRule(i, 5)
+            applyRule(i, 7)
+        }
+
+        if (output.length === 0) {
+            console.log(i)
+        } else if (checkDivisor(i, 17)) {
+            console.log(output.reverse().join(""))
```

Comment:
I feel like this part should be with the rest of rules rather than baked into the "output" part of code. Especially if we decided to add another rule to the mix it would be weird to try and add it if it needed to happen _after_ the reversal.

I think just taking this out and doing the reverse rule after all the other rules makes sense, then doing the string return would make sense :D

Reply:
Thanks, this makes sense! It's a helpful reminder for me to think about how easily the code can be changed in the future.